neo/app/assets/javascripts/index.js

```diff
@@ -90,15 +90,246 @@ app.controller('SearchCtrl', function($scope, $window, Asterix) {
+    $scope.controls.custom.push(info);
+  }
+
+  // load geoJson
+  function loadGeoJsonFiles() {
+    $http.get("assets/data/state.json")
+      .success(function (data) {
+        $scope.geojson.state = {
+          data: data,
+          style: $scope.styles.stateStyle
+        };
+      })
+      .error(function (data) {
+        console.log("Load state data failure");
+      });
+    // $http.get("../../../public/data/county.json")
+    //   .success(function (data) {
+    //     $scope.geojson.county = data;
+    //   })
+    //   .error(function (data) {
+    //     console.log("Load county data failure");
+    //   });
+    // $http.get("../../../public/data/city.json")
+    //   .success(function (data) {
+    //     $scope.geojson.city = data;
+    //   })
+    //   .error(function (data) {
+    //     console.log("Load city data failure");
+    //   });
+
+  }
+
+  /**
+   * Update map based on a set of spatial query result cells
+   * @param    [Array]     mapPlotData, an array of coordinate and weight objects
+   */
+  function drawMap(result) {
+    var maxWeight = 10;
+    var minWeight = 0;
+
+    // find max/min weight
+    angular.forEach(result, function (value, key) {
+      maxWeight = Math.max(maxWeight, value);
+    });
+
+    var range = maxWeight - minWeight;
+    if (range < 0) {
+      range = 0
+      maxWeight = 0
+      minWeight = 0
+    }
+    if (range < 10){
+      range = 10
+    }
+
+    var colors = $scope.legend.colors;
+
+    function getColor(d) {
+      return d > minWeight + range * 0.9 ? colors[10] :
+        d > minWeight + range * 0.8 ? colors[9] :
+          d > minWeight + range * 0.7 ? colors[8] :
+            d > minWeight + range * 0.6 ? colors[7] :
+              d > minWeight + range * 0.5 ? colors[6] :
+                d > minWeight + range * 0.4 ? colors[5] :
+                  d > minWeight + range * 0.3 ? colors[4] :
+                    d > minWeight + range * 0.2 ? colors[3] :
+                      d > minWeight + range * 0.1 ? colors[2] :
+                        d > minWeight ? colors[1] :
+                          colors[0];
+    }
+
+    function style(feature) {
+      return {
+        fillColor: getColor(feature.properties.count),
+        weight: 2,
+        opacity: 1,
+        color: 'white',
+        dashArray: '3',
+        fillOpacity: 0.5
+      };
+    }
+
+    // update states count
+    if($scope.geojson.hasOwnProperty('state')) {
+      angular.forEach($scope.geojson.state.data.features, function (d) {
+        if (d.properties.count)
+          d.properties.count = 0;
+        for (var k in result) {
+          if (k == d.properties.name)
+            d.properties.count = result[k];
+        }
+      });
+
+      // draw
+      $scope.geojson.state.style = style;
+    }
+  }
+
   $scope.$watch(
     function() {
       return Asterix.mapResult;
     },
 
     function(newResult) {
       $scope.result = newResult;
+      if(Object.keys($scope.result).length != 0) {
```

Comment:
I want to check if a javascript object is empty, that's the way I find from the internet